massgov/openmass
# DP-20331: Fix duplicate records in content views

**Description:**
 - Change the manual select and insert to a merge .
 - Add a unique constraint on mass_superset_data.nid .

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-20331


**To Test:**
 - Run ahoy updatedb
 - Comment out the if condition that checks $last_run, $in_window, etc around line 30 of mass_superset.module . This will allow the cron to run outside of the normal time window that's specified in /admin/config/services/superset .
 - Run ahoy drush core:cron which will run the cron and process some of the items in the superset queue.
 - Wait approximately five minutes for some of the queue items to be processed.
 - Rebuild the cache with ahoy drush cache:rebuild.
 - View the pages below to ensure that pageviews and a GPA score exists for some of the data.
 - Verify that there are no duplicate records.

**URLs**
- /admin/home
- /admin/content?node_org_filter=Massachusetts%20Digital%20Service%20%2844371%29%20-%20Organization&score=3&order=pageviews&sort=desc&last_updated=All
- /admin/content
- /admin/ma-dash/my-content
- /admin/ma-dash/needs-review

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
